exercises/etl/src/example.c

```diff
@@ -0,0 +1,47 @@
+#include "etl.h"
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdio.h>
+
+static int compare(const void *a, const void *b)
+{
+   new_map *p = (new_map *) a;
+   new_map *q = (new_map *) b;
+   return (p->key - q->key);
+}
+
+int convert(legacy_map * input, int input_len, new_map ** output)
+{
+   if (input == NULL || input_len <= 0)
+      return 0;
+
+   int len = 0;
+   int i;
+   for (i = 0; i < input_len; i++)
```

Comment:
This and the later `for` loop could probably become `for (size_t i = 0; ...`
Additionally I would be for adding braces to wrap the single operation in the `for` block. I'd be interested in thoughts on this though as it is an almost purely stylistic concern. Whichever is decided could be added to the style guide maybe.
EDIT: Just realized also that if the indexing variable declaration is moved into the initialization condition of the `for`, then the `for` will _have_ to have braces to create the necessary block scope.

Reply:
+1 for moving the declaration into the `for`

Reply:
@wolf99 - Can you please elaborate a bit more about the "have to have braces". I followed you suggestion, bit still can't follow why this is a must.

Reply:
@vlzware IIRC (and I very well may not, so advised to test, I don't have an environment that I can test this myself just ATM): if the index variable (`i`) declaration is moved into the initialization condition of the `for` loop as suggested, then the main idea is that the scope of that indexing variable is limited to the body of the `for`. If the `for`doesn't have braces it doesn't have a body _as far as scope is concerned_ (i.e. no block scope is created for it to exist within). Because you use `i` with the operation within the `for` loop instead of just within the for conditions only, then it needs to be in scope, thus you need braces to create the block scope.

Reply:
@wolf99 This does not seem to be true.
But as far as stylistic preferences go, I have no problem in using braces :)